internal/subscription/core_test.go

```diff
@@ -307,12 +308,107 @@ func TestCore_DeleteSubscription(t *testing.T) {
 			mockProjectCore.EXPECT().ExistsWithID(gomock.Any(), sub.ExtractedSubscriptionProjectID).Times(1).Return(!test.wantErr, nil)
 			mockRepo.EXPECT().Exists(gomock.Any(), gomock.Any()).MaxTimes(1).Return(true, nil)
 			mockRepo.EXPECT().Delete(gomock.Any(), gomock.Any()).MaxTimes(1).Return(nil)
+			mockTopicCore.EXPECT().DeleteSubscriptionTopic(ctx, gomock.Any()).Return(nil).AnyTimes()
 			err := core.DeleteSubscription(ctx, &sub)
 			assert.Equal(t, test.wantErr, err != nil)
 		})
 	}
 }
 
+func TestCore_DeleteSubscriptionTopics_Failure(t *testing.T) {
+	ctx := context.Background()
+	ctrl := gomock.NewController(t)
+	mockProjectCore := pCore.NewMockICore(ctrl)
+	mockTopicCore := tCore.NewMockICore(ctrl)
+	mockRepo := repo.NewMockIRepo(ctrl)
+	mockBrokerStore := mocks.NewMockIBrokerStore(ctrl)
+	subscriptionName := "projects/test-project/subscriptions/test-sub"
+	projectID := "test-project"
+	subTopic := "projects/test-project/topics/test-topic"
+	subName := "test-sub"
+	topicName := "test-topic"
+	subscription := getSubModelWithInput(subscriptionName, subTopic, projectID, projectID, subName, topicName)
+	core := NewCore(mockRepo, mockProjectCore, mockTopicCore, mockBrokerStore)
+	type args struct {
+		ctx            context.Context
+		subscription   *Model
+		deleteErrTopic string
+	}
+	tests := []struct {
+		name    string
+		args    args
+		wantErr bool
+	}{
+		{
+			name: "DeleteSubscriptionTopic with Empty topic",
+			args: args{
+				ctx:            ctx,
+				subscription:   getSubModelWithInput(subscriptionName, "", projectID, projectID, subName, topicName),
+				deleteErrTopic: "",
+			},
+			wantErr: true,
+		},
+		{
+			name: "DeleteSubscriptionTopic of Retry topic",
+			args: args{
+				ctx:            ctx,
+				subscription:   subscription,
+				deleteErrTopic: subscription.GetRetryTopic(),
+			},
+			wantErr: true,
+		},
+		{
+			name: "DeleteSubscriptionTopic of Internal topic",
+			args: args{
+				ctx:            ctx,
+				subscription:   subscription,
+				deleteErrTopic: subscription.GetSubscriptionTopic(),
+			},
+			wantErr: true,
+		},
+		{
+			name: "DeleteSubscriptionTopic of DLQ topicc",
+			args: args{
+				ctx:            ctx,
+				subscription:   subscription,
+				deleteErrTopic: subscription.GetDeadLetterTopic(),
+			},
+			wantErr: true,
+		},
+		{
+			name: "DeleteSubscriptionTopic of Delay topic",
+			args: args{
+				ctx:            ctx,
+				subscription:   subscription,
+				deleteErrTopic: subscription.GetDelayTopics()[0],
+			},
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			mockProjectCore.EXPECT().ExistsWithID(gomock.Any(), tt.args.subscription.ExtractedSubscriptionProjectID).DoAndReturn(func(arg0 context.Context, arg1 string) (bool, error) {
+				if arg1 == tt.args.subscription.ExtractedSubscriptionProjectID {
+					return true, nil
+				} else {
+					return false, errors.New("project not found")
+				}
+			}).AnyTimes()
```

Comment:
These tests doesn't cover the correctness end to end as we as we are checking only the err retired. 

Please add test cases to assert `deleteSubscriptionTopic`  is called the with **right topic name** for subscription, retry, delay and dlq

Reply:
@subintp right topic name test case is already ```TestCore_DeleteSubscription``` written, i this test case i am only checking if we get any error while deleting any subscription related topics

Reply:
1. `TestCore_DeleteSubscription` tests what is happening inside that method. 
2.  `deleteSubscriptionTopic` is called multiple times from `deleteSubscriptionTopics` with multiple topic names. The correctness of `deleteSubscriptionTopics`  depends on whether we are passing the right topic name to `deleteSubscriptionTopic` method which is not covered in the test cases.

Reply:
+1. We never test which topics get deleted in the first place. This is the core functionality that needs to get tested. How do we know if we are deleting all relevant topics here?

Reply:
Added test case to test right names